### ISSUE TITLE
Support Khadas VIM3

### ISSUE THREAD

**Issue by hdelatorre** [siderolabs/talos#2909]
https://www.khadas.com/vim3

**Comment by andrewrynhard** [comment]
Wow this thing looks nice :+1: . Will look into this.

**Comment by andrewrynhard** [comment]
@hdelatorre I haven't had success building u-boot for amlogic boards in the past. Our CI uses native compiling environments and the Amlogic docs are for cross compiling. Their `Makefile` makes this assumption unfortunately. If you have any ideas, happy to hear them.

https://github.com/u-boot/u-boot/blob/896cc5aa4a8fc0c28036b9615a37f0034addad44/doc/board/amlogic/khadas-vim3.rst

**Comment by andrewrynhard** [comment]
Actually, let me try without cross compiling. Will let you know.

**Comment by andrewrynhard** [comment]
Yep, so the cross compiling is not an issue, but we run into another problem here. 

```
#55 21.43 In file included from ../include/linux/compiler.h:54,
#55 21.43                  from ../include/part_efi.h:16,
#55 21.43                  from ../include/part.h:278,
#55 21.43                  from ../include/common.h:80:
#55 21.43 ../include/linux/compiler-gcc.h:114:1: fatal error: linux/compiler-gcc10.h: No such file or directory
#55 21.43   114 | #include gcc_header(__GNUC__)
#55 21.43       | ^~~~~
#55 21.43 compilation terminated.
#55 21.44 make[3]: *** [../scripts/Makefile.autoconf:64: include/autoconf.mk] Error 1
#55 21.45 make[2]: *** [../Makefile:475: silentoldconfig] Error 1
#55 21.45 make[1]: *** No rule to make target 'include/config/auto.conf', needed by 'include/config/uboot.release'.  Stop.
#55 21.45 make: *** [Makefile:147: sub-make] Error 2
```

I believe u-boot landed support for gcc10 relatively recently. The Khadas u-boot branch did not backport this.

**Comment by MAHDTech** [comment]
Anyone get Talos to work om the VIM3?

**Comment by frezbo** [comment]
> Anyone get Talos to work om the VIM3?

please take a look at this: https://github.com/siderolabs/overlays/

could be easier now with overlays

**Comment by rothgar** [comment]
Is this something we could make an overlay for? It looks like they have uboot instructions https://docs.khadas.com/products/sbc/vim3/development/linux/build-linux-uboot

**Comment by Mershab99** [comment]
Would love to help get this over the goalpost!

Fortuitous timing, I wanted to convert my Khadas into a Control Plane node!
